packages/truffle-decoder/lib/decode/storage.ts

```diff
@@ -269,7 +351,7 @@ export default async function decodeStorageReference(definition: DecodeUtils.Ast
           }
 
           result.members[memberName] =
-            await decode(definition.typeName.valueType, valuePointer, info, web3, contractAddress);
+            await decode(valueDefinition, valuePointer, info, web3, contractAddress);
         }
       }
```

Comment:
gosh, changes to this file are sure getting difficult to read about. it all seems reasonable though. we should think about getting this more organized so it doesn't require as large a brain-cache size to look at PRs.